diff --git a/node_modules/@sveltejs/kit/.bun-tag-ed16d31cd59bfc5a b/.bun-tag-ed16d31cd59bfc5a
new file mode 100644
index 0000000000000000000000000000000000000000..e69de29bb2d1d6434b8b29ae775ad8c2e48c5391
diff --git a/src/exports/vite/index.js b/src/exports/vite/index.js
index fe829e73e240d283f9f7557ffc60ebfe6973b1b6..fc666106d4603154edd848ca73223ef00ce54b0b 100644
--- a/src/exports/vite/index.js
+++ b/src/exports/vite/index.js
@@ -575,7 +575,8 @@ async function kit({ svelte_config }) {
 		enforce: 'pre',
 
 		async resolveId(id, importer, options) {
-			if (importer && !importer.endsWith('index.html')) {
+			const could_be_server_only = id.startsWith('.') || id.startsWith('/') || id.startsWith('$') || id.startsWith('\0');
+			if (importer && !importer.endsWith('index.html') && !importer.startsWith(normalized_node_modules) && could_be_server_only) {
 				const resolved = await this.resolve(id, importer, { ...options, skipSelf: true });
 
 				if (resolved) {
